### PR TITLE
pybind11: python package added

### PR DESCRIPTION
**Pull request by henryiii**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Supporting #68150 by adding the Python utilities to the formula.

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks for working on this, @henryiii.

File: Formula/pybind11.rb [hunk 1]
```diff
@@ -47,5 +60,8 @@ def install
     python_flags = `#{Formula["python@3.9"].opt_bin}/python3-config --cflags --ldflags --embed`.split
     system ENV.cxx, "-O3", "-shared", "-std=c++11", *python_flags, "example.cpp", "-o", "example.so"
     system Formula["python@3.9"].opt_bin/"python3", "example.py"
+
+    system Formula["python@3.9"].opt_bin/"python3", "-m", "pybind11", "--includes"
+    system opt_bin/"pybind11-config", "--includes"
```
Comment (carlocab) [inline Formula/pybind11.rb:65]: Should we use an `assert_match` here to verify the output of this test?

Comment (henryiii) [reply Formula/pybind11.rb:65]: We could, it outputs `-I/usr/local/Cellar/python@3.9/3.9.1_4/Frameworks/Python.framework/Versions/3.9/include/python3.9 -I/usr/local/Cellar/pybind11/2.6.1/libexec/lib/python3.9/site-packages/pybind11/include` currently (and there's a small change we might swap `-I` for `-isystem` someday). Maybe the best thing to just verify that `libexec/site_packages` is in it?

Comment (carlocab) [reply Formula/pybind11.rb:65]: Yea, `assert_match "libexec/site_packages", ...` sounds good.

Comment (carlocab) [reply Formula/pybind11.rb:65]: Actually, should we also not try something like `import pybind11`, at least?

Comment (henryiii) [reply Formula/pybind11.rb]: Is there any way that `python -m pybind11` would work but `import pybind11` would not? Added tests for output.

Comment (carlocab) [reply Formula/pybind11.rb]: Good point; probably not. This is good enough then.

File: Formula/pybind11.rb [hunk 2]
```diff
@@ -44,8 +57,17 @@ def install
...
     EOS
 
+    version = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
+    site_packages = "lib/python#{version}/site-packages"
+
     python_flags = `#{Formula["python@3.9"].opt_bin}/python3-config --cflags --ldflags --embed`.split
     system ENV.cxx, "-O3", "-shared", "-std=c++11", *python_flags, "example.cpp", "-o", "example.so"
     system Formula["python@3.9"].opt_bin/"python3", "example.py"
+
+    test_module = shell_output("#{Formula['python@3.9'].opt_bin/'python3'} -m pybind11 --includes")
```
Comment (carlocab) [inline Formula/pybind11.rb:67]:
```suggestion
    test_module = shell_output("#{Formula["python@3.9"].opt_bin/"python3"} -m pybind11 --includes")
```
This is inside an interpolation so you're safe to use `"` here. Same below

File: Formula/pybind11.rb [hunk 3]
```diff
@@ -44,8 +57,17 @@ def install
...
+    assert_match (libexec/site_packages).to_s, test_module
+
+    test_script = shell_output("#{opt_bin/'pybind11-config'} --includes")
```
Comment (carlocab) [inline Formula/pybind11.rb:70]:
```suggestion
    test_script = shell_output("#{opt_bin/"pybind11-config"} --includes")
```

**Comment by carlocab** [comment]
Failure on Big Sur:
```
Error: 1 failed steps!
brew test --retry --verbose networkit
```
This is a known failure: https://github.com/Homebrew/homebrew-core/issues/66450

**Review by carlocab — APPROVED** [review]
Thanks again, @henryiii!

**Pull request closed without merging**